packages/@lwc/synthetic-shadow/src/shared/event-target.ts

```diff
@@ -29,20 +51,17 @@ export function getEventListenerWrapper(fnOrObj: unknown) {
     let wrapperFn = EventListenerMap.get(fnOrObj);
     if (isUndefined(wrapperFn)) {
         wrapperFn = function (this: EventTarget, event: Event) {
+            const currentTarget = eventCurrentTargetGetter.call(event) as EventTarget;
+            const actualTarget = getActualTarget(event);
+
             if (process.env.NODE_ENV !== 'production') {
-                const currentTarget = eventCurrentTargetGetter.call(event);
                 assert.invariant(
                     isFalse(isHostElement(currentTarget)),
                     'This routine should not be used to wrap event listeners for host elements and shadow roots.'
                 );
             }
 
-            const { composed } = event;
-
-            // TODO [#2121]: We should also be filtering out other non-composed events at this point
```

Comment:
Update the PR description to signal that this PR once merge should close #2121.

Reply:
Updated